test/backend/CMakeLists.txt

```diff
@@ -33,9 +33,16 @@ execute_process(
   ERROR_QUIET
 )
 if (${PYTEST_XDIST_FOUND} EQUAL 0)
-  message(STATUS "Parallel backend tests  : ON")
-  set(BACKEND_TEST_COMMAND "pytest")
-  set(BACKEND_TEST_ARGS "--forked" "-n" "$\{NPROC\}")
+  if (NOT DEFINED NPROC)
+    if (DEFINED ENV{NPROC})
+      set(NPROC $ENV{NPROC} CACHE STRING "Number of processors for pytest" FORCE)
+    else()
+      set(NPROC "auto" CACHE STRING "Number of processors for pytest" FORCE)
+    endif()
+  endif()
+  message(STATUS "Parallel backend tests  : ON (Using ${NPROC} thread(s))")
+  set(BACKEND_TEST_COMMAND "${Python3_EXECUTABLE}" "-m" "pytest")
+  set(BACKEND_TEST_ARGS "--forked" "-n" "${NPROC}")
```

Comment:
If we just use a constant then I'd think `$\{NPROC:-4\}` would be sufficient? Note the backslashes make NPROC **not** to be interpreted by cmake but rather by shell at runtime.

Reply:
The problem is that doesn't evaluate to an environment variable across platforms (i.e. Windows) [the {} format for non-lists isn't even handled uniformly across unix shells]. I agree that losing the evaluation at runtime is a disadvantage but I don't see a workaround better than handling it in cmake at generator time.

Reply:
So out of all the archs we build on, Windows is the only one that can't handle `${NPROC:-4}`? Then I'm wondering if it's better to just special case Windows. It's a bit of an eye sore having to write multiple ifs just to set an env var. Actually, do we even run pytest on Windows?

Reply:
Also on 2nd thought about `auto` vs a constant (e.g., 4), I think it's OK to use auto. Since inside a Jenkins CI docker environment, NPROC will also be set. So `auto` will only apply when you run pytest manually.

Reply:
I think you're right. This file is going to require significant overhaul to work on Windows, and worrying about this particular case is not valuable. If the ${var} format works across all other platforms we care about, that seems reasonable. I'll restore the auto per your comment as well.